specs/Standards/Wallets/InjectedWallets.md

```diff
@@ -0,0 +1,369 @@
+
+## Specification
+
+Injected wallets are browser extensions that implement the `Wallet` API (see below) on the `window` object. To avoid namespace collisions and easily detect when they're available, wallets will mount under their own key within `window.near` (e.g. `window.near.sender`).
+
+At the core of a wallet are [`signTransaction`](#signtransaction) and [`signTransactions`](#signtransactions). These methods, when given a [`Transaction`](https://nomicon.io/RuntimeSpec/Transactions) instance from [`near-api-js`](https://github.com/near/near-api-js), will prompt the user to sign with a key pair previously imported (with the assumption it has [`FullAccess`](https://nomicon.io/DataStructures/AccessKey) permission).
+
+In most cases, a dApp will need a reference to an account and associated public key to construct a [`Transaction`](https://nomicon.io/RuntimeSpec/Transactions). The [`connect`](#connect) method helps solve this issue by prompting the user to select one or more accounts they would like to make visible to the dApp. When at least one account is visible, the wallet considers the dApp [`connected`](#connected) and they can access a list of [`accounts`](#accounts) containing an `accountId` and `publicKey`.
+
+For dApps that often sign gas-only transactions, [`FunctionCall`](https://nomicon.io/DataStructures/AccessKey#accesskeypermissionfunctioncall) access keys can be added/deleted for one or more accounts using the [`signIn`](#signin) and [`signOut`](#signout) methods. While this functionality could be achieved with [`signTransactions`](#signtransactions), it suggests a direct intention that a user wishes to sign in/out of a dApp's smart contract.
+
+### Wallet API
+
+Below is the entire API for injected wallets. It makes use of `near-api-js` to enable interoperability with dApps that will already use it for constructing transactions and communicating with RPC endpoints.
+
+```ts
+import { transactions, utils } from "near-api-js";
+
+interface Account {
+  accountId: string;
+  publicKey: utils.PublicKey;
+}
+
+interface Network {
+  networkId: string;
+  nodeUrl: string;
+}
+
+interface SignInParams {
+  permission: transactions.FunctionCallPermission;
+  accounts: Array<Account>;
+}
+
+interface SignOutParams {
+  accounts: Array<Account>;
+}
+
+interface SignTransactionParams {
+  transaction: transactions.Transaction;
+}
+
+interface SignTransactionsParams {
+  transactions: Array<transactions.Transaction>;
+}
+
+interface Events {
+  accountsChanged: { accounts: Array<Account> };
+  networkChanged: { network: Network };
+}
+
+type Unsubscribe = () => void;
+
+interface Wallet {
+  id: string;
+  connected: boolean;
+  network: Network;
+  accounts: Array<Account>;
```

Comment:
Why use an array over an object hash here?

Reply:
It was the simplest way to allow wallet implementors to be able to control which account would be 'first' in the list of accounts to sign in as or sign transactions for :)